nselib/msrpctypes.lua

```diff
@@ -4523,6 +4523,26 @@ function unmarshall_SERVICE_STATUS(data, pos)
 end
 
 
+--- Unmarshalls a null-terminated Unicode string based upon a 32-bit offset (LPTSTR)
+-- @param arguments The data being processed
+-- @param startpos  The current position within the data
+-- @return The new position
+-- @return The string with null removed
+function unmarshall_lptstr(arguments, startpos)
+
+  local endpos = startpos
+  local s = ""
+
+  while s ~= "\0\0" do
```

Comment:
Looks like an endless loop if `arguments` does not contain `\0\0`

Reply:
LPTSTR datatype will have a NULL character but yeah, I will add a check for size as well.

Reply:
Be careful about making assumptions on data that are not under your control. Technically this would qualify as a DoS weakness in nmap.

Reply:
Thanks for the suggestion but I tested the script well and I didn't find any starting position of the packet, hence I ignored to add the check and as per my knowledge Windows response **will** have a NULL byte at the end of the response packet.